client/blocks/comment-detail/index.jsx

```diff
@@ -197,14 +206,26 @@ export class CommentDetail extends Component {
 }
 
 const mapStateToProps = ( state, ownProps ) => {
-	// TODO: replace with
+	// TODO: replace `const comment = ownProps.comment;` with
 	// `const comment = ownProps.comment || getComment( ownProps.commentId );`
 	// when the selector is ready.
-	const comment = ownProps.comment;
+	const {
+		comment,
+		siteId,
+	} = ownProps;
+
+	const postId = get( comment, 'post.ID' );
 
 	// TODO: eventually it will be returned already decoded from the data layer.
 	const postTitle = decodeEntities( get( comment, 'post.title' ) );
 
+	const commentsTree = getPostCommentsTree( state, siteId, postId, 'all' );
+	const parentCommentId = get( commentsTree, `[${ comment.ID }].data.parent.ID`, 0 );
+	const parentComment = get( commentsTree, `[${ parentCommentId }].data`, {} );
```

Comment:
The second arg doesn't need to be a string. eg `get( commentsTree, [ parentCommentId, 'data' ], {} );